src/UI/EmailAutofill.js

```diff
@@ -44,18 +47,27 @@ ${includeStyles}
             }
         }
         this.registerClickableButton(this.usePersonalButton, () => {
-            this.associatedForm.autofillEmail(formatDuckAddress(this.addresses.personalAddress))
+            this.fillForm(this.addresses.personalAddress)
         })
         this.registerClickableButton(this.usePrivateButton, () => {
-            this.associatedForm.autofillEmail(formatDuckAddress(this.addresses.privateAddress))
+            const email = this.addresses.privateAddress
             this.interface.refreshAlias()
+            this.fillForm(email)
         })
 
         // Get the alias from the extension
         this.interface.getAddresses().then(this.updateAddresses)
 
         this.init()
     }
+    fillForm (address) {
+        const formattedAddress = formatDuckAddress(address)
+        if (isTopFrame) {
```

Comment:
I think we can move this to interface.inboundCredential which would remove the reliance on this.associatedForm